pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintStrategyTable.java

```diff
@@ -29,6 +29,9 @@
 public class PinotHintStrategyTable {
   public static final String INTERNAL_AGG_INTERMEDIATE_STAGE = "aggIntermediateStage";
   public static final String INTERNAL_AGG_FINAL_STAGE = "aggFinalStage";
+  // Hint to denote that aggregation is to be run as a single stage rather than split into an intermediate and a final
+  // stage
+  public static final String INTERNAL_IS_SINGLE_STAGE_AGG = "isSingleStageAgg";
```

Comment:
@walterddr this will make more sense once you look at this PR: https://github.com/apache/pinot/pull/10846
We cannot leverage the final stage hint because we need to differentiate between the intermediate stage actually running in the v1 engine as a leaf stage vs. intermediate stage running in the v2 engine after performing operations such as JOIN (agg on top of join vs. agg in leaf)